waspc/README.md

```diff
@@ -17,150 +17,135 @@ If you would like to make your first contribution, here is a handy checklist we
 
-Main result of building the project is `wasp` executable (also referred to as CLI), which is both Wasp compiler, CLI and Wasp project runner in one - one tool for everything Wasp-related.
+Main result of building the project is `wasp` executable (also referred to as CLI), which is both Wasp compiler, CLI and Wasp project runner in one - one main tool for everything Wasp-related.
 
-`wasp` executable takes `.wasp` files and `ext/` dir as input and generates a web app from them.
-
-![Wasp compilation](images/wasp-diagram.png)
-
-It can then also run that web app for you, deploy it (not yet but that is coming), and manage it in other ways.
+Note that while `waspc` is the name of this Haskell project, the executable's name is `wasp` when released, and `wasp-cli` during development.
 
 ## Basics
 
-### Setup
-
-We use the [`cabal`](https://cabal.readthedocs.io/) command line tool to build
-the project. The best way to install it is via
-[ghcup](https://www.haskell.org/ghcup/). The `ghcup tui` terminal user interface
-is a convenient way of installing and selecting versions of `cabal`, `hls` and
-`ghc`.
-
-:warning: On Mac, we recommend using the official [ghcup](https://www.haskell.org/ghcup/) installer over Homebrew, as it works out of the box.
-
-Make sure to use `cabal` CLI version >=3.12, since we use some of the newer cabal functionalities.
-
-In [cabal.project](cabal.project) we're explicitly saying what compiler version
-to build Wasp with.
+### Repo
 
-```cabal
-with-compiler: ghc-x.y.z
-```
+Clone this repo (or its fork) to your machine.
 
-Please ensure that you have this ghc version installed.
+Position yourself in this directory (`waspc/`) and make sure that you are on the `main` branch.
 
-### Repo
+### `run` script
 
-Fork this repo and clone the fork to your machine (or clone this repo directly if you don't plan to contribute but just want to try it out).
+[./run](run) script captures the most common commands / workflows for developoment of `waspc` (similar to the role that `scripts` in `package.json` play in `npm` projects), e.g. `./run build`, `./run test`, ... .
 
-Position yourself in this directory (`waspc/`) and make sure that you are on the `main` branch.
+It serves both as a convenient way to easily run common development tasks and also as a sort of a documentation on how we develop `waspc`. You will see us mentioning it often in the rest of this README.
 
-### Build
+Running `./run` without any arguments will print help/usage, which is a good way to discover the typical `waspc` development workflows.
 
-```
-cabal build
-```
+> [!TIP]
+> To make it easy to run the `./run` script from any place in your waspc codebase, you can create a bash alias that points to it, e.g.:
+> ```sh
+> alias wrun="/home/martin/git/wasp-lang/wasp/waspc/run"
+> ```
 
-to build the library and `wasp` executable.
+### Setup
```

Comment:
I split the setup into 1. setting up Haskell stuff and 2. setting up npm stuff. I also simplified it a bit. I think this makes it very easy to understand and do now.